examples/Core/ThreadOverhead.cxx

```diff
@@ -127,7 +127,7 @@ int main( int argc, char * argv[] )
 
   const char * timingsFileName = argv[1];
   const int iterations = (argc>2) ? atoi( argv[2] ): 500;
-  const int threads = (argc>3) ? atoi( argv[3] ) : itk::MultiThreader::GetGlobalDefaultNumberOfThreads();
+  const int threads = (argc>3) ? atoi( argv[3] ) : itk::MultiThreaderBase::GetGlobalDefaultNumberOfThreads();
```

Comment:
Dzenan,

I think we need a 5.0.1 version of ITK and an ifdef here.

After this change is submitted, it would prevent historical performance testing.  The PerformanceBenchmarking currently conditionally compiles successfully against master and 4.12.0 (and probably older versions too).

I'd like to keep that capability so that we can test historical versions as far back as ITK will compile with C++11 compilers.

Hans

Reply:
In that case, we could wait with merging until refactoring is complete. We would then have
```cxx
#if !defined ( ITK_LEGACY_REMOVE )
#define MultiThreader MultiThreaderBase
#endif
```
which would enable PerformanceBenchmarking to work for 5.0.0 and earlier, with `ITK_LEGACY_REMOVE` set to OFF - which is the default.

And then PerformanceBenchmarking could use version>5.0.0 in its ifdef for directly referencing `MultiThreaderBase`.

Reply:
You could just use some typedef defined in ITK like `itk::ProcessObject::MultiThreaderType`.

Reply:
I like Brad's suggestion a lot!

Reply:
Wasn't `itk::ProcessObject::MultiThreaderType` added just prior to 4.13.0 release? If so, it would not be helpful for going back through history past that point.

Reply:
@dzenanz That is correct.

What is the version compatibility goal? What is the current compatibility? Is it tested? Are we prepared to continue to test and validate compatibility with older ITK's?

Reply:
The goal is to go as far back into the past as possible. This should be at least a few minor releases back (e.g. back to 4.9, or back to 4.6 or something similar).

Reply:
But using `itk::MultiThreader` for 4.x-, and `itk::ProcessObject::MultiThreaderType` for 5.x+ would be simple, and it could work right now, and without querying `ITK_LEGACY_REMOVE` define.